packages/ember-routing/lib/location/history_location.js

```diff
@@ -208,14 +207,14 @@ export default EmberObject.extend({
   onUpdateURL(callback) {
     this._removeEventListener();
 
-    this._popstateHandler = () => {
+    this._popstateHandler = run.bind(this, function() {
       // Ignore initial page load popstate event in Chrome
-      if (!popstateFired) {
-        popstateFired = true;
+      if (!this.__popstateFired) {
+        this.__popstateFired = true;
         if (this.getURL() === this._previousURL) { return; }
       }
       callback(this.getURL());
```

Comment:
is it possible `callback` contains the run? If so, we may need to cleanup that side.

Reply:
tests generally don't navigate with the url right?

Reply:
Confirm.